src/main/java/ca/uhn/fhir/jpa/starter/OAuth2Helper.java

```diff
@@ -0,0 +1,123 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Base64;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+
+public class OAuth2Helper {
+	private static final Logger logger = LoggerFactory.getLogger(OAuth2Helper.class);
+	private static final String OAUTH_ENABLED = System.getenv("OAUTH_ENABLED");
+
+	protected Boolean isOAuthEnabled() {
```

Comment:
Please move these two functions into the interceptor so that the responsibility for "checking if auth is enabled" is located there for both authentication methods.